### PR TITLE
[Final] [Javi] Timer fixes

### PR DESCRIPTION
**Pull request by tonycosentini**

### CODE REVIEW

**Comment by JaviSoto** [comment]
Will need re-merge.

File: Zinc/ZincRepo.m [hunk 1]
```diff
@@ -1003,6 +1004,8 @@ - (void) updateBundleWithID:(NSString*)bundleID taskRef:(ZincTaskRef*)taskRef
                 [taskRef addDependency:task];
                 [self addOperation:taskRef];
             }
+        } else {
```
Comment (JaviSoto) [inline Zinc/ZincRepo.m]: I didn't add this :-/

File: Zinc/ZincRepo.m [hunk 2]
```diff
@@ -1003,6 +1004,8 @@ - (void) updateBundleWithID:(NSString*)bundleID taskRef:(ZincTaskRef*)taskRef
...
+            taskRef.completionBlock();
```
Comment (tonycosentini) [inline Zinc/ZincRepo.m]: Whoops, this wasn't meant to be in here.

There does seem to be an issue where you update a bundle that already has the newest version and the completionBlock isn't called. I'll make a separate pull.

**Comment by JaviSoto** [comment]
cc @amrox

**Comment by JaviSoto** [comment]
:shipit:

**Pull request merged**